sdk/python/lib/pulumi/resource.py

```diff
@@ -654,8 +662,9 @@ def __init__(self,
 
             opts.aliases = opts.aliases.copy()
             for parent_alias in opts.parent._aliases:
-                opts.aliases.append(inherited_child_alias(
-                    name, opts.parent._name, parent_alias, t))
+                child_alias = inherited_child_alias(
+                    name, opts.parent._name, parent_alias, t)
+                opts.aliases.append(cast(Output[Union[str, Alias]], child_alias))
```

Comment:
what *does* mypy think the type of child_alias is here?

Reply:
`Output[str]`